examples/sources/src/main/java/com/hortonworks/streamline/examples/sources/FileReaderSpout.java

```diff
@@ -0,0 +1,223 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+public class FileReaderSpout extends BaseRichSpout {
+    protected static final Logger LOG = LoggerFactory.getLogger(FileReaderSpout.class);
+    private static final String FIRST_NAME = "firstName";
+    private static final String LAST_NAME = "lastName";
+    private transient SpoutOutputCollector spoutOutputCollector;
+    private final String path;
+    private String delimiter;
+    private String outputStream;
+
+    public FileReaderSpout (String path) {
+        this.path = path;
+    }
+
+    public FileReaderSpout withDelimiter (String delimiter) {
+        if (delimiter == null || delimiter.isEmpty()) {
+            String errorString = "Delimiter must not be empty";
+            LOG.error(errorString);
+            throw new RuntimeException(errorString);
+        }
+        this.delimiter = delimiter;
+        return this;
+    }
+
+    public FileReaderSpout withOutputStream (String outputStream) {
+        if (outputStream == null || outputStream.isEmpty()) {
+            String errorString = "Output stream must not be empty";
+            LOG.error(errorString);
+            throw new RuntimeException(errorString);
+        }
+        this.outputStream = outputStream;
+        return this;
+    }
+
+    @Override
+    public void declareOutputFields(OutputFieldsDeclarer outputFieldsDeclarer) {
+        outputFieldsDeclarer.declareStream(outputStream, new Fields(StreamlineEvent.STREAMLINE_EVENT));
+    }
+
+    @Override
+    public void open(Map map, TopologyContext topologyContext, SpoutOutputCollector spoutOutputCollector) {
+        LOG.info("Opening FileReaderSpout");
+        this.spoutOutputCollector = spoutOutputCollector;
+    }
+
+    @Override
+    public void nextTuple() {
+        BufferedReader br = null;
+        try {
+            String line;
+            br = new BufferedReader(Files.newBufferedReader(Paths.get(path), StandardCharsets.UTF_8));
+            while ((line = br.readLine()) != null) {
+                String[] result = line.split(delimiter);
+                if (result.length != 2) {
+                   LOG.error("Format of input file not as expected. Expecting {} separated first name and last name", delimiter);
+                } else {
+                    Map<String, Object> event = new HashMap<>();
+                    event.put(FIRST_NAME, result[0].trim());
+                    event.put(LAST_NAME, result[1].trim());
+                    List<Object> values = new ArrayList<>();
+                    values.add(new MyStreamlineEvent(event, outputStream));
```

Comment:
Why do we require user to construct a `MyStreamlineEvent` class?. They should be able to use the default implementation which we also use internally.

Reply:
StreamlineEventImpl class that we use internally is part of streamline-common module. I believe we only want the user to have dependency on streamline-sdk. Moreover, when you first created StreamlineEvent, it was an interface rather than a non final class that can be extended. I wanted to showcase how one can provide their own implementation as well. If we want to force the user to use our internal class, then we probably dont need to expose it as an interface. Do you remember why it was chosen to be an interface? Either ways, there is nothing preventing them from using the internal implementation since streamline-layout-storm dependency brings streamline-common transitively.

Reply:
I was suggesting to use the default impl mainly to simplify things for the first time user. They will copy the sample code and keep adding their logic and may violate the immutability requirement in the event which will lead to issues later. 

So instead of MyStreamlineEvent it could be something like:

```java
        StreamlineEventImpl.builder()
                .put(FIRST_NAME, result[0].trim())
                .put(LAST_NAME, result[1].trim())
                .build();
```

Regarding keeping it in the SDK, yes we should provide the default implementation in the SDK itself and rename the class to DefaultStreamlineEvent. Can you raise a followup issue for this?

Main idea to keep it as an interface was to only specify the contract and provide maximum flexibility for implementation within streamline itself (and maybe for some advanced users). E.g. Currently we have StreamlineEventImpl extending the Map interface which would not be possible if StreamlineEvent was an Abstract class.

Reply:
Will use StreamlineEventImpl in this example then. Have raised https://github.com/hortonworks/streamline/issues/965